src/database/SsiCredentialIssuer.DbAccess/Repositories/CompanySsiDetailsRepository.cs

```diff
@@ -317,8 +317,18 @@ public IAsyncEnumerable<CredentialExpiryData> GetExpiryData(DateTimeOffset now,
             .ToAsyncEnumerable();
     }
 
-    public void RemoveSsiDetail(Guid companySsiDetailId, string bpnl, string userId) =>
+    public void RemoveSsiDetail(Guid companySsiDetailId, string bpnl, string userId)
+    {
+        var assignments = context.CompanySsiDetailAssignedDocuments
+            .Where(x => x.CompanySsiDetailId == companySsiDetailId).ToList();
+
+        var documents = context.Documents.Where(document => assignments.Select(assign => assign.DocumentId).Contains(document.Id)).ToList();
+
+        context.CompanySsiDetailAssignedDocuments.RemoveRange(assignments);
+        context.Documents.RemoveRange(documents);
+
         context.CompanySsiDetails.Remove(new CompanySsiDetail(companySsiDetailId, bpnl, default, default, bpnl, userId, DateTimeOffset.MinValue));
```

Comment:
this should be done in the business logic. 

please create a repository method to receive the needed Ids of the documents. which that you can create new instances and remove them. than you don't need to receive the complete documents and companySsiDetailAssignedDocuments.